agent/consul/state/catalog_test.go

```diff
@@ -1799,7 +1799,7 @@ func TestStateStore_EnsureService_VirtualIPAssign(t *testing.T) {
 	require.NoError(t, s.EnsureService(10, "node1", ns1))
 
 	// Make sure there's a virtual IP for the foo service.
-	vip, err := s.VirtualIPForService(structs.ServiceName{Name: "foo"})
+	vip, err := s.VirtualIPForService(structs.PeeredServiceName{ServiceName: structs.ServiceName{Name: "foo"}})
```

Comment:
Can you add a basic test to ensure the peering separation is enforced? Similar to what we often test for partitions/namespaces.

Could register "foo" as a local service and an imported service, then verify that yields two distinct virtual IPs.